pageserver/src/page_service.rs

```diff
@@ -38,18 +38,22 @@ use crate::PageServerConf;
 use crate::ZTenantId;
 use crate::ZTimelineId;
 
+use postgres_ffi::pg_constants;
+
 // Wrapped in libpq CopyData
 enum PagestreamFeMessage {
     Exists(PagestreamRequest),
     Nblocks(PagestreamRequest),
     Read(PagestreamRequest),
+    DbSize(PagestreamRequest),
```

Comment:
Hm, I feel uneasy about it. This call is not the part of smgr api and you mix it in. Db size is sum of all the relation sizes and you have a loop iterating through it in the pageserver. You may move that loop in the postgres and use existing nblocks and that would be it. Of course it would be slower due to network roundtrips. If you want to avoid that you can just add a separate top-level command in the pageserver and call it in a separate connection. @hlinnaka WDYT?

Reply:
It seems fine to me. Yeah, you could loop in postgres side too, but I don't see anything wrong with doing it in the pageserver either.

> If you want to avoid that you can just add a separate top-level command in the pageserver and call it in a separate connection.

That seems fine to me, too. I don't have a preference between doing that and what this PR does.